calour/ms1_experiment.py

```diff
@@ -0,0 +1,168 @@
+'''
+ms1 experiment (:mod:`calour.ms1_experiment`)
+=======================================================
+
+.. currentmodule:: calour.ms1_experiment
+
+Classes
+^^^^^^^^
+.. autosummary::
+   :toctree: generated
+
+   Ms1Experiment
+'''
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--,  Calour development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from logging import getLogger
+
+import numpy as np
+
+from .experiment import Experiment
+from .util import _get_taxonomy_string, _to_list
+
+
+logger = getLogger(__name__)
+
+
+class Ms1Experiment(Experiment):
+    '''This class contains the data for a Mass-Spec ms1 spectra experiment or a meta experiment.
+
+    Parameters
+    ----------
+    data : :class:`numpy.ndarray` or :class:`scipy.sparse.csr_matrix`
+        The abundance table for OTUs, metabolites, genes, etc. Samples
+        are in row and features in column
+    sample_metadata : :class:`pandas.DataFrame`
+        The metadata on the samples
+    feature_metadata : :class:`pandas.DataFrame`
+        The metadata on the features
+    description : str
+        name of experiment
+    sparse : bool
+        store the data array in :class:`scipy.sparse.csr_matrix`
+        or :class:`numpy.ndarray`
+
+    Attributes
+    ----------
+    data : :class:`numpy.ndarray` or :class:`scipy.sparse.csr_matrix`
+        The abundance table for OTUs, metabolites, genes, etc. Samples
+        are in row and features in column
+    sample_metadata : :class:`pandas.DataFrame`
+        The metadata on the samples
+    feature_metadata : :class:`pandas.DataFrame`
+        The metadata on the features
+    exp_metadata : dict
+        metadata about the experiment (data md5, filenames, etc.)
+    shape : tuple of (int, int)
+        the dimension of data
+    sparse : bool
+        store the data as sparse matrix (scipy.sparse.csr_matrix) or numpy array.
+    description : str
+        name of the experiment
+
+    See Also
+    --------
+    Experiment
+    '''
+    def __repr__(self):
+        '''Return a string representation of this object.'''
+        return 'Ms1Experiment %s with %d samples, %d features' % (
+            self.description, self.data.shape[0], self.data.shape[1])
+
+    def prepare_gnps(self):
```

Comment:
cannot do it in init since values are initialized after the init in the read_openms() function.
Added to automatically call this function if gnps_file is supplied in read_openms
(user does not need to do anything)